apps/nameguard.io/src/app/docs/displayed-name/page.tsx

```diff
@@ -0,0 +1,274 @@
+                  getExampleDisplayedName(DisplayedNameExample.LONG_NAME).name
+                }
+              />
+            </div>
+          </div>
+        </div>
+        <div className="divide-y divide-gray-100">
+          <div className="grid grid-cols-2 gap-x-6 py-5">
+            <div className="w-full flex justify-center items-center font-mono">
+              <pre>
+                {"<Text name \n displayTooltipWhenNameOverflows={false} />"}
+              </pre>
+            </div>
+
+            <div className="flex items-center justify-center">
+              <Text
+                displayTooltipWhenTextOverflows={false}
+                text={
+                  getExampleDisplayedName(DisplayedNameExample.WITHOUT_TOOLTIP)
+                    .name
+                }
+              />
+            </div>
+          </div>
+        </div>
+        <div className="divide-y divide-gray-100">
+          <div className="grid grid-cols-2 gap-x-6 py-5">
+            <div className="w-full flex justify-center items-center font-mono">
+              <pre>{"<Text name \n maxDisplayWidth={60} />"}</pre>
+            </div>
+
+            <div className="flex items-center justify-center">
+              <Text
+                maxDisplayWidth={60}
+                text={
+                  getExampleDisplayedName(DisplayedNameExample.SHORT_MAX_WIDTH)
+                    .name
+                }
+              />
+            </div>
+          </div>
+        </div>
+        <h2 className="text-lg font-bold w-full text-center border-b-2 border-t-2 py-4 mt-10">
+          {"<"}DisplayedName {"/>"} component
+        </h2>
+        <div className="divide-y divide-gray-100">
+          <div className="grid grid-cols-2 gap-x-6 py-5 text-center font-medium">
+            <div className="font-semibold">Component usage</div>
+            <div className="font-semibold">Example</div>
+          </div>
+        </div>
+
+        <div className="divide-y divide-gray-100">
+          <div className="grid grid-cols-2 gap-x-6 py-5">
+            <div className="w-full flex justify-center items-center font-mono">
+              <pre>{"<DisplayedName name />"}</pre>
+            </div>
+
+            <div className="flex items-center justify-center">
+              <DisplayedName
+                name={getExampleDisplayedName(DisplayedNameExample.SHORT_NAME)}
+              />
+            </div>
+          </div>
+        </div>
+        <div className="divide-y divide-gray-100">
+          <div className="grid grid-cols-2 gap-x-6 py-5">
+            <div className="w-full flex justify-center items-center font-mono">
+              <pre>
+                {
+                  "<DisplayedName name\n textStylingClasses='text-purple-400' />"
+                }
+              </pre>
+            </div>
+
+            <div className="flex items-center justify-center">
+              <DisplayedName
+                textStylingClasses="text-purple-400"
```

Comment:
Can you please include an example in the docs showing what happens if we set custom text styling classes like these AND there's an overflow tooltip?

I want to confirm that the styling of the text in the tooltip will NOT be changed if we change the styling of the name itself. In other words, maybe I want the ENS name to appear in a larger font or in a different color, etc.. but then I don't want anything to change about the styling in the tooltip.

What's a good way for us to achieve this?

Reply:
@FrancoAguzzi Not sure about adding the complexity of that enum. I think it's ok to allow UI components to be given class names that don't make sense. Maybe needs two distinct custom classname fields?